pkg/cli/tail_test.go

```diff
@@ -211,6 +227,180 @@ Validation failed for 1 logs (out of 2)
 	}
 }
 
+func TestStreamTailCommand(t *testing.T) {
+	t.Parallel()
+
+	ct := time.Now().UTC()
+
+	validLog := &loggingpb.LogEntry{
+		InsertId: "test-log",
+		Payload: &loggingpb.LogEntry_JsonPayload{
+			JsonPayload: &structpb.Struct{
+				Fields: map[string]*structpb.Value{
+					"service_name":  structpb.NewStringValue("foo_service"),
+					"method_name":   structpb.NewStringValue("foo_method"),
+					"resource_name": structpb.NewStringValue("foo_resource"),
+					"authentication_info": structpb.NewStructValue(&structpb.Struct{
+						Fields: map[string]*structpb.Value{
+							"principal_email": structpb.NewStringValue("foo@bet.com"),
+						},
+					}),
+				},
+			},
+		},
+		Labels: map[string]string{"environment": "dev", "accessing_process_name": "foo_apn"},
+	}
+
+	bs, err := protojson.Marshal(validLog)
+	if err != nil {
+		t.Fatalf("failed to mashal log to JSON: %v", err)
+	}
+	validLogJSON := stripSpaces(string(bs))
+
+	cases := []struct {
+		name            string
+		args            []string
+		puller          *fakePuller
+		expFilter       string
+		expOut          string
+		expErrSubstr    string
+		expStderrSubstr string
+	}{
+		{
+			name: "success_stream_tail",
+			args: []string{"-scope", "projects/foo", "-f"},
+			puller: &fakePuller{
+				logEntries: []*loggingpb.LogEntry{{}},
+			},
+			expFilter: fmt.Sprintf(
+				`LOG_ID("audit.abcxyz/unspecified") OR `+
+					`LOG_ID("audit.abcxyz/activity") OR `+
+					`LOG_ID("audit.abcxyz/data_access") OR `+
+					`LOG_ID("audit.abcxyz/consent") OR `+
+					`LOG_ID("audit.abcxyz/system_event") `+
+					`AND timestamp >= %q`,
+				ct.Add(-2*time.Hour).Format(time.RFC3339),
+			),
+			expOut:       `{}`,
+			expErrSubstr: "stream tail validate cancelled",
+		},
+		{
+			name: "success_stream_tail_validate",
+			args: []string{
+				"-scope", "projects/foo",
+				"-duration", "4h",
+				"-additional-filter", `resource.type = "gae_app" AND severity = ERROR`,
+				"-validate",
+				"-f",
+			},
+			puller: &fakePuller{
+				logEntries: []*loggingpb.LogEntry{validLog},
+			},
+			expFilter: fmt.Sprintf(
+				`LOG_ID("audit.abcxyz/unspecified") OR `+
+					`LOG_ID("audit.abcxyz/activity") OR `+
+					`LOG_ID("audit.abcxyz/data_access") OR `+
+					`LOG_ID("audit.abcxyz/consent") OR `+
+					`LOG_ID("audit.abcxyz/system_event") `+
+					`AND timestamp >= %q AND resource.type = "gae_app" `+
+					`AND severity = ERROR`,
+				ct.Add(-4*time.Hour).Format(time.RFC3339),
+			),
+			expOut: fmt.Sprintf(`%s
+Successfully validated log (InsertId: "test-log")
+
+Validation failed for 0 logs (out of 1)
+`, validLogJSON),
+			expErrSubstr: "stream tail validate cancelled",
+		},
+		{
+			name: "stream_tail_fail",
+			args: []string{
+				"-scope", "projects/foo",
+				"-override-filter", "test-filter",
+				"-f",
```

Comment:
with -v, the output will be different, should we test this case as well?

Reply:
I think it's a `stream_tail_validate_fail` case covers this?